crux-agent/app/api/routers/jobs.py

```
@@ -21,6 +21,158 @@
+        
+        # Sort by creation date (newest first) and limit results
+        jobs.sort(key=lambda x: x.created_at or datetime.min.replace(tzinfo=timezone.utc), reverse=True)
+        
+        return jobs[:limit]
+        
+    except Exception as e:
+        logger.error(f"Error listing jobs: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to list jobs: {str(e)}",
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)